guides/docs/testing/testing_controllers.md

```diff
@@ -133,169 +222,489 @@ defmodule HelloWeb.Router do
 Let's build a test around what we think a successful `show/2` should look like.
 
 ```elixir
-test "Responds with a newly created user if the user is found" do
-  user = User.changeset(%User{}, %{name: "John", email: "john@example.com"})
-  |> Repo.insert!
+# test/hello_phoenix_web/controllers/user_controller_test.exs line 41 (or so)
 
-  response = build_conn()
-  |> get(user_path(build_conn(), :show, user.id))
+test "Responds with user info if the user is found", %{conn: conn, user1: user} do
+  response = conn
+  |> get(user_path(conn, :show, user.id))
   |> json_response(200)
 
-  expected = %{ "data" => %{ "name" => "John", "email" => "john@example.com" } }
+  expected = %{
+    "data" =>
+    %{ "email" => user.email, "name" => user.name }
+
+  }
 
   assert response == expected
 end
 ```
 
-This is very similar to our `index/2` test, except `show/2` requires a user id, and our data is a single JSON object instead of an array.
+This is very similar to our `index/2` test, except `show/2` requires a
+user id, and our data is a single JSON object instead of an
+array. Notice that because we only need one user for this test, we're
+only grabbing `user1:` from the context, and mapping it to `user`.
+
 
-When we run our test tells us we need a `show/2` action.
+When we run our test tells us we need a `HelloWeb.UserController.show/2` action.
 
 ```elixir
+# lib/hello_web/controllers/user_controller.ex
+
 defmodule HelloWeb.UserController do
   use HelloWeb, :controller
-
-  alias Hello.{Accounts.User, Repo}
+  alias Hello.Accounts
 
   def index(conn, _params) do
-    users = Repo.all(User)
-    render conn, "index.json", users: users
+    users = Accounts.list_users()
+    render(conn, "index.json", data: users)
   end
 
   def show(conn, %{"id" => id}) do
-    case Repo.get(User, id) do
-      user -> render conn, "show.json", user: user
-    end
+    user = Accounts.get_user!(id)
+    render conn, "show.json", data: user
   end
+
 end
 ```
 
-You'll notice we only handle the case where we successfully find a user. When we TDD we only want to write enough code to make the test pass. We'll add more code when we get to the error handling test for `show/2`.
+You might notice the exclamation point in the `get_user!/1` function.
+This convention means that this function will throw an error if the
+requested user is not found.  You'll also notice that we aren't
+properly handling the possibility of a thrown error here. When we TDD we only want to write enough code to make the test pass. We'll add more code when we get to the error handling test for `show/2`.
 
 Running the test tells us we need a `render/2` function that can pattern match on `"show.json"`:
 
 ```elixir
 defmodule HelloWeb.UserView do
   use HelloWeb, :view
 
-  def render("index.json", %{users: users}) do
-    %{data: render_many(users, HelloWeb.UserView, "user.json")}
+  def render("index.json", %{data: users}) do
+    %{data:
```

Comment:
The formatting for the old version was fine, we can just add the `"show.html"` clause here.

Reply:
fixed